### PR TITLE
[incubator/storm] Apache Storm

### PR DESCRIPTION
**Pull request by jorwalk**
**What this PR does / why we need it**:
This pull request adds the necessary templates and manifest files to deploy Apache Storm with Helm.

### CODE REVIEW

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please compare with what you get from `helm create` using Helm 2.8+ and update accordingly.

**Comment by jorwalk** [comment]
/assign @unguiculus 

Thank you for the recommendation to use `helm create`. I have updated the pull request to reflect the changes necessary.

Updated to `apiVersion: apps/v1beta2`

Added  metadata.labels.app to each deplpyment
```
labels:
    app: {{ template "storm.nimbus.name" . }}
```
Added spec.selector.matchLabels to each deployment
```
selector:
    matchLabels:
      app: {{ template "storm.nimbus.name" . }}
      release: {{ .Release.Name }}
```

File: incubator/storm/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+name: storm
+version: 1.0.0
+keywords:
+  - storm
+  - zookeeper
+home: http://storm.apache.org/
+sources:
+  - https://github.com/apache/storm
+maintainers:
+  - name: Allison Richardet
```
Comment (unguiculus) [inline incubator/storm/Chart.yaml]: Use Github username

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please update `_helpers.tpl` and `chart` labels as well.

File: incubator/storm/Chart.yaml (hunk 1 above)
Comment (jorwalk) [reply incubator/storm/Chart.yaml]: Thanks for the feedback, I have replaced the name with the Github username.

**Comment by jorwalk** [comment]
Sorry to be dense. Could you please provide more guidance as to what changes you're referring to with updating `_helpers.yaml` and `chart` labels as well?

Are you talking about adding this to the _helpers file?
```
{{/*
Create chart name and version as used by the chart label.
*/}}
{{- define "storm.chart" -}}
{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
{{- end -}}
```

**Comment by unguiculus** [comment]
Yes, add this and use it.

**Comment by jorwalk** [comment]
@unguiculus Thank you for the clarification. 

ci/circleci build failure:
```
You are using pip version 9.0.3, however version 10.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
--2018-06-19 14:39:46--  https://github.com/mikefarah/yaml/releases/download/1.13.1/yaml_linux_amd64
Resolving github.com (github.com)... 192.30.253.112, 192.30.253.113
Connecting to github.com (github.com)|192.30.253.112|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2018-06-19 14:39:46 ERROR 404: Not Found.

Exited with code 8
```

File: incubator/storm/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,75 @@
+{{/* vim: set filetype=mustache: */}}
```
Comment (unguiculus) [inline incubator/storm/templates/_helpers.tpl:1]: This is not based on the latest `_helpers.tpl` . Please update.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/storm/templates/_helpers.tpl (hunk 2 above)
Comment (jorwalk) [reply incubator/storm/templates/_helpers.tpl:1]: @unguiculus - Running version 2.9.1 of helm, when I exec the command `helm create` in the terminal, that line is still included in the generated `_helpers.tpl`.  

When evaluating other charts in both stable and incubator folders, I still see that line included in all of them.  

* [stable/Artifactory](https://github.com/kubernetes/charts/blob/master/stable/artifactory/templates/_helpers.tpl)
* [stable/datadog](https://github.com/kubernetes/charts/blob/master/stable/datadog/templates/_helpers.tpl)

Could you please advise as to the update that you are looking to see?

Comment (unguiculus) [reply incubator/storm/templates/_helpers.tpl:1]: This line is not a problem. Please update the fullname templates.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Pull request closed without merging**